### PR TITLE
Show unread message count on lobby chat tabs.

### LINKED ISSUE #9833: Lobby chat tabs should show unread message count
Right now there is no way to tell when new messages come in.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27861754-lobby-chat-tabs-should-show-unread-message-count?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by pchote**
Closes #9833.

### CODE REVIEW

**Comment by RoosterDragon** [comment]
If I join the global chat from the multiplayer tab, then switch to the skirmish tab, all messages seen on the multiplayer tab are regarded as unread.

File: OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs [hunk 1]
```diff
@@ -62,6 +62,10 @@ enum ChatPanelType { Lobby, Global }
 		readonly LabelWidget chatLabel;
 		bool teamChat;
 
+		int lobbyChatUnreadMessages;
+		int globalChatLastReadMessages;
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs]: Making this `static` fixes the thing that RoosterDragon mentioned. I think I'd like it better that way, too.

You just need a way to reset it to 0 when you press the `leave chat` button.

Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs]: Setting the initial value in the lobbylogic ctor should be an even easier fix

Comment (obrakmann) [inline OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs]: Derp. Yeah, it is.

**Comment by pchote** [comment]
Fixed.

**Comment by penev92** [comment]
:+1:

**Comment by obrakmann** [comment]
Works nicely now. :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/2db89075bbe535542f5d84625aa4edbfa791588d)

**Pull request merged**